common/metrics/config_test.go

```diff
@@ -160,3 +161,54 @@ func (s *MetricsSuite) TestSetDefaultPerUnitHistogramBoundaries() {
 		s.Equal(test.expectResult, config.PerUnitHistogramBoundaries)
 	}
 }
+
+func TestMetricsHandlerFromConfig(t *testing.T) {
+	t.Parallel()
+
+	logger := log.NewTestLogger()
+
+	for _, c := range []struct {
+		name         string
+		cfg          *Config
+		expectedType interface{}
+	}{
+		{
+			name:         "nil config",
+			cfg:          nil,
+			expectedType: &noopMetricsHandler{},
+		},
+		{
+			name: "tally",
+			cfg: &Config{
+				Prometheus: &PrometheusConfig{
+					Framework:     FrameworkTally,
+					ListenAddress: "localhost:0",
+				},
+			},
+			expectedType: &tallyMetricsHandler{},
+		},
+		{
+			name: "opentelemetry",
+			cfg: &Config{
+				Prometheus: &PrometheusConfig{
+					Framework:     FrameworkOpentelemetry,
+					ListenAddress: "localhost:0",
+				},
+			},
+			expectedType: &otelMetricsHandler{},
+		},
+	} {
+		c := c
+		t.Run(c.name, func(t *testing.T) {
+			t.Parallel()
+
+			handler, err := MetricsHandlerFromConfig(logger, c.cfg)
+			require.NoError(t, err)
+			defer func() {
```

Comment:
I doubt it matters but you may want t.Cleanup here rather than defer. Very low importance